From 6d575d98565ce3119a14359eecb11ccdff92a303 Mon Sep 17 00:00:00 2001
From: Yash Shinde <53660251+Yashinde145@users.noreply.github.com>
Date: Thu, 29 Jun 2023 18:10:15 +0530
Subject: [PATCH] Fix function inline errors in debug optimization (-Og)

Compiler does not inline any functions when using debug optimization (-Og).
Hence, remove -Winline flag when compiling with debug optimization.

Signed-off-by: Nicolas Marguet <nicolas.marguet@windriver.com>
---
Upstream-Status: Backport [https://github.com/rsyslog/librelp/commit/6d575d9]

 configure.ac | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/configure.ac b/configure.ac
index 21c1fde..1204c4e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -230,6 +230,12 @@ if test "$enable_debug" = "no"; then
         AC_DEFINE(NDEBUG, 1, [Defined if debug mode is disabled.])
 fi

+#Compiler does not inline any functions when not optimizing(-Og).
+#Hence, remove -Winline flag when DEBUG is enabled.
+#ifdef DEBUG
+WARN_CFLAGS="$(echo "$WARN_CFLAGS" | sed s/-Winline//g)"
+#endif
+
 # valgrind
 AC_ARG_ENABLE(valgrind,
         [AS_HELP_STRING([--enable-valgrind],[Enable valgrind tests@<:@default=yes@:>@])],
--
2.39.0
